### PR TITLE
Add longoptions support for gmtsplit.

### LINKED ISSUE #8066: Add longoptions support for gmtsplit.
**Description of the desired feature**

Add longoptions support for gmtsplit.

**Are you willing to help implement and maintain this feature?** Yes/No

### PR DESCRIPTION
**Pull request by rbdavis**
**Description of proposed changes**

Python differences:

  No python module.

Julia differences:

  None (other than aliases).

Fixes #8066

**Reminders**

- [ ] Make sure that your code follows our style. Use the other functions/files as a basis.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Describe changes to function behavior and arguments in a comment below the function declaration.
- [ ] If adding new functionality, add a detailed description to the documentation and/or an example.

### CODE REVIEW

**Comment by PaulWessel** [comment]
Looks fine.  I changed one alias for -Q to be --fields which I think we used for similar cases (block*).

**Review by PaulWessel — APPROVED** [review]

File: src/longopt/gmtsplit_inc.h [hunk 1]
```diff
@@ -25,34 +25,16 @@ static struct GMT_KEYWORD_DICTIONARY module_kw[] = {
...
-	{ 0, 'S', "",
-	          "",                  "",
-	          "",                  "",
-		  GMT_TP_STANDARD },
+	{ 0, 'A', "azimuth_tolerance|azim_tol", "", "", "", "", GMT_TP_STANDARD },
+	{ 0, 'C', "course_change",     "", "", "", "", GMT_TP_STANDARD },
+	{ 0, 'D', "min_distance|min_dist", "", "", "", "", GMT_TP_STANDARD },
+	{ 0, 'F', "filter",            "", "", "", "", GMT_TP_STANDARD },
+	{ 0, 'N', "multifile|multi",   "", "", "", "", GMT_TP_STANDARD },
+	{ 0, 'Q', "outputs|fields",
```
Comment (joa-quim) [inline src/longopt/gmtsplit_inc.h:33]: Maybe drop the `outputs`, or at least make it `output`

File: src/longopt/gmtsplit_inc.h [hunk 2]
```diff
@@ -25,34 +25,16 @@ static struct GMT_KEYWORD_DICTIONARY module_kw[] = {
...
+	          "x,y,z,d,h",         "x,y,z,distance|dist,heading|hdg",
+	          "",                  "",
+		  GMT_TP_MULTIDIR },
+	{ 0, 'S', "full_input|dist_head", "", "", "", "", GMT_TP_STANDARD },
```
Comment (joa-quim) [inline src/longopt/gmtsplit_inc.h:37]: `full_input` is not really informative. I would drop this one too.

**Comment by PaulWessel** [comment]
I did change -S to --extended.

**Comment by joa-quim** [comment]
And with that you made it different from in GMT.jl

**Comment by PaulWessel** [comment]
Oops what does Julia use?

**Comment by joa-quim** [comment]
Not briliant, `dist_head`
(when in doubt, do ``? modulename``

**Comment by PaulWessel** [comment]
That is bad enough to deserve an alias!

**Comment by PaulWessel** [comment]
OK added _dist_head_ as an alias to match Julia but --_extended_ is the main directive name.

**Pull request merged**